File: post.go
package main

type Post struct {
	Title   string `json:"title"`
	Content string `json:"content"`
	Author  string `json:"author"`
}

type Posts struct {
	Posts []Post
}

func (u *User) NewPost(title, content string) *Post {
	return &Post{
		Title:   title,
		Content: content,
		Author:  u.Name,
	}
}

File: src/main/java/com/overclock/yuechu/entity/Post.java
package com.overclock.yuechu.entity;

import com.baomidou.mybatisplus.annotation.TableId;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.util.Date;

/**
 * @author wangyu
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Post implements Serializable {
    private static final long serialVersionUID = 1L;

    public Post(Long userId, String title, String content, Long tagId, int isDelete, Date createTime, Date updateAt) {
        this.userId = userId;
        this.title = title;
        this.content = content;
        this.tagId = tagId;
        this.isDelete = isDelete;
        this.createTime = createTime;
        this.updateAt = updateAt;
    }

    @TableId
    private Long id;

    private Long userId;

    private String title;

    private String content;

    private Long tagId;

    private int isDelete;

    private Date createTime;

    private Date updateAt;
}
